src/lib/input/input.ts

```diff
@@ -351,7 +350,26 @@ export class MatInput extends _MatInputMixinBase implements MatFormFieldControl<
    * Implemented as part of MatFormFieldControl.
    * @docs-private
    */
-  get shouldLabelFloat(): boolean { return this.focused || !this.empty; }
+  get shouldLabelFloat(): boolean {
+    if (this._elementRef.nativeElement.nodeName.toLowerCase() === 'select') {
```

Comment:
Instead of doing this in a getter, the check can be done in the constructor since the element's `nodeName` won't change.